### ISSUE TITLE
cmake broken

### ISSUE THREAD

**Issue by GitMensch** [Bill-Gray/PDCurses#116]
After restructuring the repo following wmcbrine the cmake build (and therefore also Travis + Appveyor) seems to be broken.
@jwinarske is there any option for you to have a look at this? @Bill-Gray any idea?

**Comment by Bill-Gray** [comment]
@GitMensch I know virtually nothing about about CMake.  I did look at the error logs,  and I see nothing that makes any sense to me. There _have_,  I should note,  been some quite useful errors from both Appveyor and Travis in the past,  and I've caught and crushed some bugs that way.  But this time,  I've no idea what's going on.

Appveyor claims to be using a mix of Visual Studio and OpenWATCOM flags (?).  The Travis error reads
```
CMake Error at CMakeLists.txt:23 (project):
  VERSION "...0" format invalid.
```
Nothing has actually changed in the CMakeLists.txt file since Joel provided it to us.

**Comment by jwinarske** [comment]
@GitMensch I should have some time next week to take a look.

**Comment by jwinarske** [comment]
There are missing spaces in these.  Adding back one space per line fixes it.  Longer term I'll make it ignore the space, but short term the fix would be to add the spaces back to curses.h

#define PDC_VER_MAJOR   4
#define PDC_VER_MINOR   1
#define PDC_VER_CHANGE  1

**Comment by Bill-Gray** [comment]
Ah,  now _that_ would never have occurred to me... thank you.  I did a check against the last build that went through cmake without trouble,  and it did have an extra space on each of those three lines.  Added 'em back, pushed the change,  and... Travis CI is now happy;  AppVeyor remains unconvinced.  Still,  it suggests we're on the right path.

**Comment by jwinarske** [comment]
Looking at the Appveyor log, looks like the Watcom URL may have changed, or is broken:  `Error downloading remote file: One or more errors occurred.`

I'll take a closer look in a few days.

**Comment by jwinarske** [comment]
The AppVeyor fix will be to edit the yaml file, which is un-related to CMake.

**Comment by Bill-Gray** [comment]
I was a little slow on the uptake there,  I admit.  For OpenWATCOM 2.0,  it is indeed just a moved or missing file (the compiler itself):
`https://github.com/open-watcom/travis-ci-ow-builds/archive/master.zip`
and a grep caused me to realize that's from the yaml file.  Haven't figured out yet where that file has gone.  For OpenWATCOM 1.9,  the compiler is found and used,  and everything compiles until you get to the executables.  Then there's a mysteriously broken line,  which I suspect means another spacing issue?  With MSVC,  there's a similarly mysteriously broken line.
I think we have a pile-up of issues here.  AppVeyor last completed successfully on 2019 July 4.  On the subsequent commit,  some builds failed;  a few commits later,  all AppVeyor builds failed.
As you may have gathered,  this is outside my area of expertise... but I'm getting the impression that the automated builds may be rather delicate things,  if adding/removing whitespace in C code breaks them.  Is that a fair assessment?  Can we expect them to fail frequently after making seemingly unimportant changes to the source code?  I like the idea of having this sort of testing (very useful with a project of this ilk that is highly cross-platform),  but it's not so helpful if it's prone to "false alarm" fails.

**Comment by jwinarske** [comment]
I would expect periodic maintenance with any automated testing.  One option to eliminated the dependency would be to host the toolchain on your own server.  I think the regex can be adjusted to be agnostic to white space.

**Comment by GitMensch** [comment]
I've fixed the OpenWatcom URL in caf2b021cd9aaa374fa067878614c1b00d84ab50 and the broken cmake part after removing the icon was fixed by 3e43378e5ac303df33657b8e2d1899fe2c50b0ea.

As mentioned in #90 there are still some cmake related issues, and as @jwinarske asked this is now assigned to him - thank you for this.

Things I've spotted:

* the wincon port looks for version stuff that is missing
* the watcom builds are all broken in the linking phase:
> C:\projects\pdcurses\demos\ptest.c: 309 lines, included 5643, 0 warnings, 0 errors
cd C:\projects\pdcurses\build
[ 75%] Linking C executable ptest.exe
cd C:\projects\pdcurses\build\os2
wlink   system os2v2 name ptest.exe  debug watcom all file {'CMakeFiles\os2_ptest.dir\__\demos\ptest.c.obj' }  librarypdcursesstatic.lib
Open Watcom Linker Version 2.0 beta May 15 2020 00:41:14 (64-bit)
Copyright (c) 2002-2020 The Open Watcom Contributors. All Rights Reserved.
Portions Copyright (c) 1985-2002 Sybase, Inc. All Rights Reserved.
Source code is available under the Sybase Open Watcom Public License.
See http://www.openwatcom.org/ for details.
Error! E3033: directive error near 'librarypdcursesstatic.lib'
Error(E42): Last command making (os2\ptest.exe) returned a bad status
* additional the options should be adjusted to include `-q`, from the toolset manual:
> 4.11.11 Operate Quietly - "q" Option
The "q" option suppressing the banner and copyright notice that is normally displayed when the OpenWatcom Library Manager is invoked
Example: `wlib -q -l mylib`

There are possible more issues... In general it would be nice to have a cmake support table like this in #90 (and working again).

**Comment by jwinarske** [comment]
Looks like there are some Watcom issues - https://github.com/open-watcom/open-watcom-v2/issues/599

Once I get through them, I will lock the toolchain to a commit.  Otherwise it will just keep breaking.

**Comment by jmalak** [comment]
#125

**Comment by jwinarske** [comment]
After the Watcom update it looks like only VS2015/VS2017 x86 builds are broken. I suspect it happened when someone updated the cmake version without testing impact.

[Appveyor build status (Windows)](https://ci.appveyor.com/project/jwinarske/pdcurses-21dj3)
[Travid build status (Mac/Linux)](https://travis-ci.com/github/jwinarske/PDCurses)

**Comment by GitMensch** [comment]
@jwinarske do yo mind checking the cmake issues [with Appveyor](https://help.appveyor.com?

**Comment by jwinarske** [comment]
looking at it... @GitMensch

**Comment by jwinarske** [comment]
I have a fix.  Once the build matrix is complete I'll make a PR.

**Comment by GitMensch** [comment]
Seems to be fixed with the current builds - feel free to request a re-opening.
I'd still like to see a build matrix in some new file in this repo enabling a fast review on the state of tests (and an x11 test in travis).
